readme.md

```diff
@@ -41,6 +43,67 @@ au User asyncomplete_setup call asyncomplete#register_source({
     \ })
 ```
 
+#### Basic go to definition usage
+
+Once you have configured this plug-in following the steps above, create a file
+`test.nim` with the following contents:
+
+```
+import strutils
+
+proc testMe() =
+  echo "test".toLower
+
+proc main() =
+  testMe()
+  test_me()
+
+when isMainModule:
+  main()
+```
+
+Put your cursor on the first `testMe()` call inside the `main()` proc, and
+while still in normal mode, type the default binding to jump to the definition
+of the `testMe()` proc, which is `gd`. If the plugin is correctly installed and
+nimsuggest is working, your cursor should jump to the sixth column of the third
+line, where the proc is defined. Type `Ctrl+o` to jump back, and repeat the
+same on the line below: since Nim syntax is case insensitive, the go to
+definition command will work the same on the `test_me()` call.
+
+Now position your cursor on the `toLower` call at the end of the fourth line
+and repeat the go to definition command. Nimsuggest works across source files
+and into Nim's own standard library, so you should end up looking at the
+`toLower*(s: string)` proc, which at the moment of writing this is located on
+[line 1546 of the lib/pure/unicode.nim
+file](https://github.com/nim-lang/Nim/blob/aa072b952555804e516861d9ed62846e341f7938/lib/pure/unicode.nim#L1546).
```

Comment:
I would prefer to have this annotated with `[1]` and have the link referenced at the end of the file.